### DISCUSSION TITLE
question about the data in performance .ini files

### DISCUSSION THREAD

**Discussion by DutchBigEd** [probonopd/MiniDexed#515] [Q&A] [answered]
if I look at the data in the performance .ini files I see a lot of parameters that makes sense to me, they have a name and a value that mostly are logically to me, but one sticks out and that is the VoiceData.
That one has a lot (156) of variables and I'm wondering if there is some info about this to read?
I can imagine it has to do with the settings of each of the 6 operators in a TG.

Thanks!

**Comment by Banana71** [comment, marked as answer]
Hey Ed,

you are trying to read the matrix :-)

The voice data is saved in HEX format in the Performance.ini. Behind it are all the parameters that can be found under Voice Edit.
https://github.com/probonopd/MiniDexed/wiki/Menu#menu-tree  
This makes it possible that the voice data (patches) can be changed individually for each part of the performance and saved in the performance.

![matrix](https://github.com/probonopd/MiniDexed/assets/104296205/6902e425-1b72-4c39-a63f-78fdf21dc11b)


![patch](https://github.com/probonopd/MiniDexed/assets/104296205/60ed287e-2e38-45ba-9eb3-1578f7d4912f)
the 156 bytes compose a DX7 patch

**Reply by DutchBigEd** [reply]
Thanks, I knew it wouldn't be easy ;)
I have worked with some Yamaha FM chips like the YMF262, they have something similar (though less complicated...)
